openff/pablo/exceptions.py

```diff
@@ -51,3 +51,15 @@ def __init__(self, serial: str, possible_indices: Collection[int]):
             msg += "but multiple corresponding ATOM/HETATM records were found "
             msg += f"(records {','.join(map(str, possible_indices))})"
         super().__init__(msg)
+
+
+class AmbiguousMatchError(PabloError):
+    """Two matches disagree where they overlap."""
+
+    def __init__(self, reasons: list[str]):
+        self.reasons = reasons
+        super().__init__(
+            f"resdef does not unambiguously map: {''.join(f'\n  {reason}' for reason in reasons)}",
+        )
+
+    pass
```

Comment:
(not blocking) This doesn't seem right, though maybe I'm missing a detail (or haven't written code in too long). 
```suggestion
```